### PR TITLE
Update oauth credentials

### PR DESCRIPTION
**Pull request by alecharp**
Because we upgraded some plugins that `oauth-credentials` plugin is using, I released a beta version so that it can be validated here.

Once this PR is validated, I'll do a "real" release.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,8 +74,6 @@
     <google.guava.version>20.0</google.guava.version>
     <oauth2.revision>151</oauth2.revision>
     <google.http.version>1.21.0</google.http.version>
-    <configuration-as-code.version>1.19</configuration-as-code.version>
```
Comment (craigdbarber) [inline pom.xml]: Why remove the configuration-as-code version prop? That seems orthogonal to the stated purpose of this PR.

File: pom.xml [hunk 2]
```diff
@@ -191,7 +199,6 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>credentials</artifactId>
-      <version>${credentials.version}</version>
```
Comment (craigdbarber) [inline pom.xml]: Why remove this property as well?

**Comment by alecharp** [comment]
@craigdbarber I removed the plugins versions to use the versions defined on the Jenkins BOM. This resolves the maven-enforcer-rule error created by the mismatch of `credentials` plugin's version. It is also a good way to make sure to use the recommended version of plugins for the Jenkins version your plugin is targeting.

**Comment by craigdbarber** [comment]
@alecharp thanks for clarifying. LGTM

**Comment by alecharp** [comment]
Sorry, I just caught that you merged the PR. I filled the PR with a `-beta` release to make sure the cleanup didn't break `google-oauth` plugin and wanted to update the version once we agreed on the changeset and validated that the content of the release was file.

I just released the `0.4`, so I'll open a new PR so you can use this version.
